lib/PipelineC/PipelineC.cpp

```diff
@@ -28,9 +28,38 @@
 using namespace pipeline;
 using namespace ::revng::pipes;
 
-void rp_set_custom_abort_hook(AbortHook Hook) {
-  setAbortHook(Hook);
-}
+/// Custom raw_ostream class that writes the result in a heap allocated char*
+class CharStarStream : public llvm::raw_ostream {
```

Comment:
This looks like early optimization that pushes us to do fragile memory management.
I'd just go with a `llvm::raw_string_ostream` and copy the data over to a newly `malloc`'d region.
Alternatively, instead of returning a raw `const char *` we could have a new type that represents a `std::string` (or a `SmallVector`, if we use `raw_svector_ostream`) that has a `rp_mytype_get_data` function and a `rp_mytype_destroy` function.